modules/core/src/lifecycle/constants.ts

```diff
@@ -5,15 +5,15 @@ export const LIFECYCLE = {
   AWAITING_GC: 'Discarded. Awaiting garbage collection',
   AWAITING_FINALIZATION: 'No longer matched. Awaiting garbage collection',
   FINALIZED: 'Finalized! Awaiting garbage collection'
-};
+} as const;
+
+export type Lifecycle = typeof LIFECYCLE[keyof typeof LIFECYCLE];
 
 /* Secret props keys */
 // Symbols are non-enumerable by default, does not show in for...in or Object.keys
 // but are copied with Object.assign ¯\_(ツ)_/¯
 // Supported everywhere except IE11, can be polyfilled with core-js
-export const PROP_SYMBOLS = {
-  COMPONENT: Symbol.for('component'),
-  ASYNC_DEFAULTS: Symbol.for('asyncPropDefaults'),
-  ASYNC_ORIGINAL: Symbol.for('asyncPropOriginal'),
-  ASYNC_RESOLVED: Symbol.for('asyncPropResolved')
-};
+export const COMPONENT: unique symbol = Symbol.for('component');
```

Comment:
Call these `COMPONENT_SYMBOL` etc since we moved them out of the object?

Reply:
Just a note, I broke these out of an object because TS does not allow me to use the `unique symbol` type in a map, and `symbol` cannot be used as index in another type (`StatefulComponentProps`).